src/Mobs/AggressiveMonster.cpp

```diff
@@ -46,6 +47,36 @@ void cAggressiveMonster::EventSeePlayer(cPlayer * a_Player, cChunk & a_Chunk)
 
 
 
+bool cAggressiveMonster::CanSeeMobType(eMonsterType a_MobType, cAggressiveMonster * a_Monster, int a_SightDistance)
+{
+	m_World->ForEachEntity([&](cEntity & a_Entity)
```

Comment:
```suggestion
	m_World->ForEachEntityInBox([&](cEntity & a_Entity, ...)
```

Maybe not check every entity. This might lead to lags if multiple locations are loaded. The box should be around the checked entity and have a radius of a_SightDistance